src/main/java/bio/terra/service/DrsService.java

```diff
@@ -174,6 +192,23 @@ private DRSContentsObject makeDrsContentsObject(FSObjectBase fsObject, String sn
         return contentsObject;
     }
 
+    private String makeHttpsFromGs(String gspath) {
+        try {
+            URI gsuri = URI.create(gspath);
+            String gsBucket = gsuri.getAuthority();
+            String gsPath = StringUtils.removeStart(gsuri.getPath(), "/");
+            String encodedPath = URLEncoder.encode(gsPath, StandardCharsets.UTF_8.toString());
```

Comment:
Does the ordering of when we do the encode here matter? It feels odd to me to encode before the `String.format`, but that might be because I'm used to python treating string literals as unicode.

If we can, I'd prefer to do the encode last.

Reply:
Not sure I understand the comment about python and Unicode. AFAIK, this doesn't have to do with the character coding.

The only part of the URL that needs encoding is the path, so that is what I encoded. I could run encode over the whole thing, but nothing in the rest of it will be changed.